zerver/lib/thumbnail.py

```diff
@@ -453,9 +453,18 @@ def rewrite_thumbnailed_images(
         path_id = image_link["href"].removeprefix("/user_uploads/")
         if to_delete and path_id in to_delete:
             # This was not a valid thumbnail target, for some reason.
-            # Trim out the whole "message_inline_image" element, since
+
+            # image has title, trim out the whole "message_inline_image" element, since
             # it's not going be renderable by clients either.
-            inline_image_div.decompose()
+            if image_link.get("title"):
```

Comment:
Whether the image tag has a title _happens_ to be aligned with if removing it will result in an empty element -- but it seems much more robust (and readable) to instead remove the inline image div, and _check if the result is empty_, since that's the bad result that we're trying to avoid.  Otherwise if in the far-away Markdown code, we decide to have a default `title=` on the image tag, then this will start producing the bad effect (an empty message).  Check for the actual behaviour that's bad, not the thing which currently causes that bad behaviour (which may change, particularly since it's far away from here).

Reply:
That wouldn't handle if we started making it `title="Image"`.  Again, we don't know what the change in the other code is going to be, and we should not be dependent on a second-order property.